src/order.cpp

```diff
@@ -3374,6 +3377,23 @@ static inline RtrBestResult decideWhereToRepairAndBalance(DROID *psDroid)
 			{
 				continue;	// cannot reach position
 			}
+			uint32_t droidsAtRepair = 0;
+			for (DROID *psOtherDroid : apsDroidLists[psDroid->player])
+			{
+				if (psDroid == psOtherDroid)
+				{
+					continue;
+				}
+				/* Search for droids within a radius of 2 tiles */
+				if (psOtherDroid->isDamaged() && droidSqDist(psOtherDroid, psStruct) < TILE_WIDTH * TILE_WIDTH * 4)
+				{
+					droidsAtRepair++;
+				}
+			}
+			if (droidsAtRepair > REPAIR_FACILITY_MAX_CAPACITY)
+			{
+				continue;
```

Comment:
Eliminating these repair facilities from contention altogether (which is what this does) seems like it could have other potentially-undesirable drawbacks.

(For example, with this change, if all repair facilities are crowded - by the metric - then none of them are candidates to even move towards. This then triggers the fallback behavior, and could (if there are no repair droids) lead to droids up for repair going all the way to the HQ. Whereas someone might want all the droids to head towards repair facilities, even if they are currently crowded.)

Reply:
> This then triggers the fallback behavior, and could (if there are no repair droids) lead to droids up for repair going all the way to the HQ.

First of all, this most likely won't happen, because if units are normally distributed to repair facilities, they will be faster to repair (at most we can increase the limit from 8 to 12, but I think it's fine as it is).
Second, as a last resort, let them go to HQ instead of crowding around the repair facilities. It can happen that a lot of damaged units will be repairing and crowding around, but suddenly an enemy comes (ex. VTOL bombers) and kills them all.
The situation can be improved by utilizing the droid repairers additionality as you've already mentioned.

> Did you not find the existing behavior that already "randomly" distributes droids amongst multiple nearby repair facilities sufficient?

No, it doesn't work well in practice, no matter how you do it you still have to make the stations close together and perpendicular to the flow for greater efficiency.

Reply:
> > Did you not find the existing behavior that already "randomly" distributes droids amongst multiple nearby repair facilities sufficient?
> 
> No, it doesn't work well in practice, no matter how you do it you still have to make the stations close together and perpendicular to the flow for greater efficiency.

But that's based on the current way the code selects which facilities are "close enough", which is something we could alternatively change.

Reply:
> which seems like something we could alternatively change.

That's what I'm suggesting...

You can make any algorithm you want, but it still has to take into account that some units are already "queued" for repair in a certain repair facility and this facility should be skipped and another nearby should be picked instead.

Reply:
![triangle](https://github.com/Warzone2100/warzone2100/assets/3750982/932b6b95-c632-43c0-bf20-323b4c69a7a2)

Here is the most basic example showing the problem. Making repair facilities close to each other creates another problem, but it solves the problem on the screenshot.